R/filter_pd.R

```diff
@@ -0,0 +1,99 @@
+#' Filter a Dataset to Only Include the Source Parameter Records up to and
+#' including the first PD. These records are passed to downstream derivations
+#' regarding responses such as BOR.
+#'
+#' @param dataset An input BDS style data set is expected
+#'
+#' @param filter_source A filter condition is expected, upon which the ADRS
+#'        data set will be filtered to capture the PARAM containing the PD
+#'        response
+#'
+#' @param source_pd The variable name which contains the dates of the responses
+#'        A date/datetime object is expected.
+#'
+#' @param source_datasets A named list of data sets is expected.
```

Comment:
@teckla-gsk @bundfussr  is there a reason we do a plural here as convention? wouldn't they only ever pick one dataset to source the PD date from?

Reply:
In other admiral functions we named the parameter `source_datasets` because more than one `date_source` object was expected. So you could have more than one source dataset.

However, here we expect only one `date_source` object and one source dataset. So we could name the parameter `source_dataset`. I am not sure what is best. `source_dataset` is more precise in the context of the function and `source_datasets` would be consistent with other admiral functions handling `date_source` or similar objects.

Reply:
let's go `source_dataset` please - i found it confusing otherwise.